Segundo entregable

carlosupper opened the pull request:
*se agrega extension para el consumo de apio para obtener la imagen
*se agrega celda customizada para mostrar las películas
*se agrega controlador para pantalla de popular movies
*se agrega network manager para el consumo de api
*se agrega modelo Movie
*se modifica el consumo que se tenia inicialmente por URLSession (se borrara ese archivo en el sig commit)
*se agregar el controlador y vista para mostrar información de película
* se agrega protocolo para el evento de tap en imagen y mostrar informacion de pelicula

CarlosNitsuga committed:
se cambio la estructura de movie agregando el protocolo codable
se cambio el consumo de la API por URLSession

CarlosNitsuga committed:
se agrega extension para el consumo de apio para obtener la imagen
se agrega celda customizada para mostrar las películas
se agrega controlador para pantalla de popular movies
se agrega network manager para el consumo de api
se agrega modelo Movie
se modifica el consumo que se tenia inicialmente por URLSession (se borrara ese archivo en el sig comió)
se agregar el controlador y vista para mostrar información de película

alejandrombWize commented on BAZProjectC1/Extensions/UIImageViewExtension.swift:
```diff
@@ -0,0 +1,30 @@
+import UIKit
+
+extension UIImageView {
+    func downloaded(from url: URL, contentMode mode: ContentMode = .scaleAspectFill){
```
`func downloaded(from url: URL, contentMode mode: ContentMode = .scaleAspectFill) {` espacio.

alejandrombWize commented on BAZProjectC1/Movies/MovieCard/MovieCardCellController.swift:
```diff
@@ -0,0 +1,45 @@
+    
+    @objc func imageTapped(tapGestureRecognizer: UITapGestureRecognizer)
+    {
+        if  (tapGestureRecognizer.view as? UIImageView) != nil {
```
Los parentesis no son necesarios.

alejandrombWize commented on BAZProjectC1/Movies/MovieInformationController.swift:
```diff
@@ -0,0 +1,26 @@
+import Foundation
+import UIKit
+
+final class MovieInformationController:UIViewController {
```
Espacio: `final class MovieInformationController: UIViewController {`

alejandrombWize commented on BAZProjectC1/Movies/MovieInformationController.swift:
```diff
@@ -0,0 +1,26 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        movieReview.isEditable = false
+        if let movies = movies, let id = id {
+            movieTitile.text = movies[id].title
+            movieReview.text = movies[id].overview
+        }
```
Tener cuidado al obtener un valor de una matriz con un indice. Podría causar un error grave en ejecución. Podría agregar alguna validación que impida que el id de un indice mayor al que puede tener movies.

alejandrombWize commented on BAZProjectC1/Movies/PopularViewController.swift:
```diff
@@ -0,0 +1,73 @@
+final class PopularViewController: UIViewController {
+    
+    @IBOutlet weak var collectionView: UICollectionView!
+    let reuseIdentifier = "moviecard"
```
Usar access control.

alejandrombWize commented on BAZProjectC1/Movies/PopularViewController.swift:
```diff
@@ -0,0 +1,73 @@
+    }
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        
+        return CGSize(width: (self.collectionView.frame.width/2)-8 , height: (self.collectionView.frame.height/3.5))
```
Espacios:
`return CGSize(width: (self.collectionView.frame.width / 2) - 8 , height: (self.collectionView.frame.height / 3.5))`

alejandrombWize reviewed (commented):
Para el entregable, documentar funciones y hacer uso de access control.

CarlosNitsuga committed:
pull request comments resolved
- add documentation
- removed and put spaces
- secure access in array (add extension)
